app/celer-sim/Transporter.cc

```diff
@@ -203,6 +204,33 @@ void Transporter<M>::accum_action_times(MapStrDouble* result) const
         }
     }
 }
+//---------------------------------------------------------------------------//
+/*!
+ * Print progress after N events when requested.
+ */
+template<MemSpace M>
+void Transporter<M>::progress(EventId const id,
+                              size_type const num_primaries) const
+{
+    CELER_EXPECT(num_primaries > 0);
+
+    auto const id_val = id.unchecked_get();
+    std::string prim = (num_primaries == 1) ? " primary" : " primaries";
+    std::string msg = "Event " + std::to_string(id_val) + ": transporting "
+                      + std::to_string(num_primaries) + prim;
```

Comment:
You're doing a lot of work that may not be output. This is also very different from how the output message is constructed: `operator<<` does nothing if the log level is suppressed, but here you're doing a lot of conversion and object construction.

Reply:
Good point. The suggested refactoring with `if (id_val % print_progress_ ==0) { print_progress(...); }` also looks cleaner.